### PR TITLE
Swift better error handling

### PR DESCRIPTION
**Pull request by grvgoel81**
## Motivation and Context
Swift better error handling

Jira Link:

## Description
* Swift better error handling.
* Previously there is generic error everytime "privateKeyNot Generated Error" but after update it gives exact error like "retrieve or import share failed" in screenshot.

## Screenshots (if appropriate):
Screenshot 2024-11-07 at 11 30 00 AM

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [ ] My code follows the code style of this project. (run lint)
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [ ] My code requires a db migration.

### CODE REVIEW

File: ios/Classes/SingleFactorAuthFlutterPlugin.swift [hunk 1]
```diff
@@ -46,7 +46,7 @@ public class SingleFactorAuthFlutterPlugin: NSObject, FlutterPlugin {
             case "init":
                 let args = call.arguments as? String
                 guard let data = args?.data(using: .utf8) else {
-                    return result(throwKeyNotGeneratedError())
+                    return result(throwParamMissingError(param: args!))
```
Comment (metalurgical) [inline ios/Classes/SingleFactorAuthFlutterPlugin.swift:49]: Forced unwrap of args here could fail if args is null, or can args never be null?

Comment (metalurgical) [reply ios/Classes/SingleFactorAuthFlutterPlugin.swift:49]: Please check other places for this as well

**Comment by metalurgical** [comment]
This is much better thanks.

Please also remove this function as well:
```
public func throwKeyNotGeneratedError() -> FlutterError
```

**Comment by grvgoel81** [comment]
> This is much better thanks.
> 
> Please also remove this function as well:
> 
> ```
> public func throwKeyNotGeneratedError() -> FlutterError
> ```

done.

**Review by chaitanyapotti — APPROVED** [review]
lgtm

**Pull request merged**